File: 2073 time-needed-to-buy-tickets/solution.cpp
class Solution {
public:
    int timeRequiredToBuy(vector<int>& ti, int k) {
        int t=0;
        for(int i=0;i<ti.size();i++) {
            if(i<=k) t+=min(ti[i],ti[k]);
            else t+=min(ti[i],ti[k]-1);
        }
        return t;
    }
};

File: 2073 time-needed-to-buy-tickets/solution.rs
impl Solution {
    pub fn time_required_to_buy(ti: Vec<i32>, k: i32) -> i32 {
        let mut t=0;
        for i in 0..ti.len() {
            if i<=k as usize{
                t+=ti[i].min(ti[k as usize]);
            } else {
                t+=ti[i].min(ti[(k as usize )]-1);
            }
        }
        t as i32
    }
}
